src/BTCPayServer.Lightning.All/LightningConnectionString.cs

```diff
@@ -292,6 +294,44 @@ private static bool TryParseNewFormat(string str, out LightningConnectionString
                             return false;
                         }
                     }
+                    break;
+                case LightningConnectionType.Eclair:
+                    
+                    
+                    var eclairserver = Take(keyValues, "server");
+                    
+                    if (eclairserver == null)
+                    {
+                        error = $"The key 'server' is mandatory for lnd connection strings";
+                        return false;
+                    }
+                    if (!Uri.TryCreate(eclairserver, UriKind.Absolute, out var eclairuri)
+                        || (eclairuri.Scheme != "http" && eclairuri.Scheme != "https"))
+                    {
+                        error = $"The key 'server' should be an URI starting by http:// or https://";
+                        return false;
+                    }
+
+                    result.BaseUri = eclairuri;
+                    result.Password = Take(keyValues, "password");
+                    result.BitcoinHost  = Take(keyValues, "bitcoin-host");
```

Comment:
what is this?

Reply:
Not sure what you mean

Eclair RPC uses a mandatory password with no username 
the bitcoin stuff is because eclair uses bitcoind's wallet so we need the rpc url and auth string to connect to it for `GetDepositAddress`